assessment/libs/complex.php

```diff
@@ -0,0 +1,636 @@
+//          default is 12 decimal places.
+//
+// Returns:
+// The agument of a complex number in radian (or degree) as a float. 
+
+function cx_arg(array $num, string $argin="rad", int $roundto=12) {
+    if (count($num)!=1) { echo "cx_arg expects 1 complex number as an input"; return "";}
+    
+    $re=$num[0][0];
+    $im=$num[0][1];
+    $r= cx_modul($num);
+    $th1=asin(abs($im/$r));
+
+    if ($re>=0 && $im>=0){
+            $theta=$th1;
+    }    
+        else if($re<0 && $im<0){
+            $theta=pi() +$th1;
+        }
+        else if($re<0 && $im>=0){
+            $theta=pi() -$th1;
+        }
+        else {
+            $theta=2*pi() -$th1;
+        }
+    if ($argin=="deg"){
+        $th2=rad2deg($theta);
+    }
+        else{
+            $th2=$theta;
+        }
+   
+    return round($th2, $roundto);
+}
+
+
+
+//-------------------------------------------------Conjugate----------------------------------------------
+// Function: cx_conj(num, [roundto = 12])
+// Returns the conjugate of a complex number: conjugate of (a+bi)-->(a-bi)
+//
+// Parameters:
+// num: An array of real and imaginary parts of a complex number given in square brackets: num = array([Re, Im]).
+// roundto: Optional - number of decimal places to which values should be rounded off; 
+//          default is 12 decimal places.
+//
+// Returns:
+// The conjugate of a complex number in an array (same format as the input).
+
+function cx_conj(array $num, int $roundto=12) {
+    if (count($num)!=1) { echo "cx_conj expects 1 complex number as an input"; return "";}
+    
+    $re=$num[0][0];
+    $im=$num[0][1];
+    $im2=-$im;
+    $st= array([round($re, $roundto),round($im2, $roundto)]); 
+   
+   return $st;
+}
+
+//------------------------------------------------Polar form --------------------------------------------
+// Function: cx_std2pol(num,[argin = "rad", roundto = 12])
+// Converts the standard form to polar form and returns the modulus and the argument as a paired value.
+//
+// Parameters:
+// num: An array of real and imaginary parts of a complex number given in square brackets: num = array([Re, Im]).
+// argin: Optional - Unit for the argument; default is "rad" for radian. For argument in degree, argin = "deg".
+// roundto: Optional - number of decimal places to which values should be rounded off; 
+//          default is 12 decimal places.
+//
+// Returns:
+// The modulus and the argument of a complex number as a paired value in an array: array([mod, arg]).
+
+function cx_std2pol(array $num, string $argin="rad", int $roundto= 12) {
+    if (count($num)!=1) { echo "cx_std2pol expects 1 complex number as an input"; return "";}
+    
+    $r= round(cx_modul($num),$roundto);
+    if ($argin=="deg"){
+        $th1=round(cx_arg($num,"deg"),$roundto);
+    }
+        else{
+            $th1=round(cx_arg($num),$roundto);
+        }
+    
+    $polar=array([$r,$th1]);    
```

Comment:
If you remove the double-arraying, the return should just be `$polar = [$r,$th1]`